src/context/layers/impact.ts

```diff
@@ -113,38 +127,39 @@ function getBaselineDataForFeature(
   );
 }
 
+const mergeFeaturesByProperty = (
+  baselineFeatures: {}[],
+  aggregateData: {}[],
+  id: string,
+  operation: string,
+) =>
+  baselineFeatures.map(feature1 => {
+    const aggregateProperties = aggregateData.find(
+      item => get(item, id) === get(feature1, ['properties', id]) && item,
+    );
+    const properties = {
+      ...get(feature1, 'properties'),
+      ...aggregateProperties,
+      impactValue: get(feature1, 'properties.data'),
+      [operation]: get(aggregateProperties, `stats_${operation}`),
+    };
+    return { ...feature1, properties };
+  });
+
 export async function fetchImpactLayerData(
   params: LayerDataParams<ImpactLayerProps>,
   api: ThunkApi,
 ) {
   const { getState, dispatch } = api;
   const { layer, extent, date } = params;
 
-  const existingHazardLayer = layerDataSelector(
-    layer.hazardLayer,
-    date,
-  )(getState());
+  const operation = layer.operation || 'median';
 
-  if (!existingHazardLayer) {
-    const hazardLayerDef = LayerDefinitions[layer.hazardLayer];
-    await dispatch(
-      loadLayerData({ layer: hazardLayerDef, extent, date } as LayerDataParams<
-        WMSLayerProps
-      >),
-    );
-  }
+  const hazardLayerDef = LayerDefinitions[layer.hazardLayer] as any;
```

Comment:
Why is this getting typed as `any`? We should be able to type it as a specific layer type...

Reply:
Still a WIP ;)

Basically, the way the layers are typed atm, hazardLayerDef does not know that it has a wcsConfig object and we don't really have the proper type readily available.